Automattic/wp-calypso
# Self-service site reset: make the UI match the mockup

Closes https://github.com/Automattic/dotcom-forge/issues/4779
Closes https://github.com/Automattic/dotcom-forge/issues/4781

**Atomic**
![image](https://github.com/Automattic/wp-calypso/assets/6851384/6acbd1c6-d600-4781-b1d9-2fca763b2a42)

**Simple**

<img width="775" alt="Screenshot 2566-12-07 at 17 40 33" src="https://github.com/Automattic/wp-calypso/assets/6851384/cfcb5811-b9de-4764-b12f-690cf542d55b">

## Proposed Changes

1. Show a list of content that will be removed
2. Only allow resetting if the domain test value matches the actual domain
3. Update the notice text to reflect the async nature of the Atomic reset
4. Refactors code so hooks now live in sub-package
5. Add a hint to the bottom of the card about Activity Log and Export https://github.com/Automattic/dotcom-forge/issues/4781
6. Make card space match mockup https://github.com/Automattic/dotcom-forge/issues/4781
7. Change text at to of card so it mentions theme changes. 

**Note**: there doesn't appear to be an easy way to detect user-installed plugins, so the plugin count includes pre-installed plugins. I think this might cause the user to panic. What do you think about just saying, "Plugins you installed"? Or maybe we can get the info, e.g., check the Activity Log instead?

## Testing Instructions



* Apply D130278-code and sandbox API
* Check out this branch
* Try with Simple and Atomic sites by visiting `/settings/start-over/:site`

This new UI needs some units tests but they can be added in https://github.com/Automattic/dotcom-forge/issues/4793

## Pre-merge Checklist
